### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by danieltimko**
- all cards sent in messages now correctly contain the card owner
- following cards should now function: Storm, Explorer
- multiple in-game bugs fixed
- Game struct removed from server_structs on client, because it'll not be needed there

### CODE REVIEW

File: server/kotc_game/src/game/column.rs [hunk 1]
```diff
@@ -32,8 +32,12 @@ impl Column {
         let number_of_cards = self.cards.len();
         if self.cards.len() >= 2 {
             let mut previous_card = self.cards.get_mut(number_of_cards-2).unwrap();
-            previous_card.revealed = true;
-            Some(previous_card.character)
+            if previous_card.revealed {
```
Comment (Puckoland) [inline server/kotc_game/src/game/column.rs:35]: Could you give me example of how can this happen? (prev card is already revealed)

File: server/kotc_game/src/game.rs [hunk 2]
```diff
@@ -253,53 +267,34 @@ impl Game {
...
-
-                let revealed_card = Rc::clone(&self.columns)
-                    .borrow_mut()
-                    .get_mut(column_index)
-                    .unwrap()
-                    .reveal_previous_card();
-                if let Some(character) = revealed_card {
-                    messages.push(self.message_update_columns());
-
-                    if character == Character::Killer {
```
Comment (Puckoland) [inline server/kotc_game/src/game.rs:267]: So Killer is now handled somewhere else?

**Review by Puckoland — APPROVED** [review]

File: server/kotc_game/src/game/column.rs (hunk 1 above)
Comment (danieltimko) [reply server/kotc_game/src/game/column.rs:35]: Killer

File: server/kotc_game/src/game.rs (hunk 2 above)
Comment (danieltimko) [reply server/kotc_game/src/game.rs:267]: Yes, it's been moved.

**Pull request merged**